src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj

```diff
@@ -20,7 +20,7 @@
 
   <ItemGroup>
     <!-- we need to keep the version of System.Reflection.Metadata in sync with dotnet/msbuild and dotnet/sdk -->
-    <PackageReference Include="System.Reflection.Metadata" Version="6.0.0" />
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataToolsetVersion)" />
```

Comment:
Currently, this is 6.0, sdk is at 7.0 and msbuild is 8.0. After this PR, I think we can do a round to update it in SDK to 8.0 first and then runtime to the same version to resolve @ViktorHofer's https://github.com/dotnet/runtime/issues/96809. (the logic is to keep the version equal or lower; preferably in sync with the version used by msbuild).

Reply:
It's fine to update this to 7.0.0 as part of this PR. Runtime always needs to wait a little bit until a version of VS is released that has the updated binding redirects for desktop msbuild. So it's good that we are lagging behind, but we shouldn't lag two releases behind.